crates/wallet/src/types.rs

```diff
@@ -37,6 +38,15 @@ impl KeychainKind {
     }
 }
 
+impl fmt::Display for KeychainKind {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            KeychainKind::External => write!(f, "external"),
+            KeychainKind::Internal => write!(f, "internal"),
```

Comment:
Maybe more clear to use uppercase?
```suggestion
            KeychainKind::External => write!(f, "External"),
            KeychainKind::Internal => write!(f, "Internal"),
```